lighthouse-core/lib/median-run.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @param {LH.Result} lhr @param {string} auditName */
+const getNumericValue = (lhr, auditName) =>
+  (lhr.audits[auditName] && lhr.audits[auditName].numericValue) || NaN;
+
+/**
+ * @param {LH.Result} lhr
+ * @param {number} medianFcp
+ * @param {number} medianInteractive
+ */
+function getMedianSortValue(lhr, medianFcp, medianInteractive) {
+  const distanceFcp = medianFcp - getNumericValue(lhr, 'first-contentful-paint');
+  const distanceInteractive =
+    medianInteractive - getNumericValue(lhr, 'interactive');
+
+  return distanceFcp * distanceFcp + distanceInteractive * distanceInteractive;
+}
+
+/**
+ * @param {Array<LH.Result>} runs
+ * @return {LH.Result|undefined}
+ */
+function computeMedianRun(runs) {
+  const validRuns = runs
+    .filter(run => Number.isFinite(getNumericValue(run, 'first-contentful-paint')))
+    .filter(run => Number.isFinite(getNumericValue(run, 'interactive')));
+  if (!validRuns.length) return undefined;
```

Comment:
a test for this?

Reply:
there are already multiple :)

Reply:
😕  i definitely see tests, but not one asserting that `computeMedianRun` sometimes returns `undefined`

Reply:
first commit :)

https://github.com/GoogleChrome/lighthouse/pull/10859/commits/3ee8e800f08901e6012344cd3ed5a36b0117f397#diff-397a8690e6788da0fa22768c7e3b5653R58-R61

Reply:
ah okay. so yesterday it was there and i was just 🙈 

but after the refactor for throwing, the method won't return undefined at all.

i must have been looking at a stale impl but up to date tests. 

weird. OKAY ALL IS GOOD thanks

Reply:
ah gotcha ya after brendan's feedback they converted to `expect().toThrow()` 👍